backend/core/src/migration/1695143897902-add-jurisdictions.ts

```diff
@@ -0,0 +1,128 @@
+  name = "addJurisdictions1695143897902"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const baseJurisdiction = (
+      await queryRunner.query(
+        `SELECT * 
+        FROM jurisdictions
+        WHERE name='Bay Area'`
+      )
+    )[0]
+    //add new jurisdictions
+    const countyArr = ["Contra Costa", "Marin", "Napa", "Santa Clara", "Solano", "Sonoma"]
+    countyArr.forEach(async (county) => {
+      const newJuris = { ...baseJurisdiction, name: county }
+      delete newJuris.id
+      delete newJuris.created_at
+      delete newJuris.updated_at
+      const jurisKeys = Object.keys(newJuris).toString()
+      const jurisValues = Object.values(newJuris)
+      await queryRunner.query(
+        `INSERT INTO jurisdictions (${jurisKeys}) 
+        VALUES (${jurisValues.map((_, index) => `$${index+1}`).toString()})`,
+        jurisValues
+      )
+    })
+
+    //link existing listings to corresponding jurisdiction
+    const existingListings = await queryRunner.query(
+      `SELECT listings.id,county
+       FROM listings 
+       LEFT JOIN address on listings.building_address_id = address.id`
+    )
+    const existingJurisdictions = await queryRunner.query(
+      `SELECT id,name
+       FROM jurisdictions`
+    )
+
+    existingListings.forEach(async (listing) => {
+      const matchingJuris = existingJurisdictions.find((juris) => juris.name === listing.county).id
+      await queryRunner.query(
+        `UPDATE listings
+        SET jurisdiction_id='${matchingJuris}'
+        WHERE id='${listing.id}'`
+      )
+    })
+
+    const existingUsersAndRoles = await queryRunner.query(
+      `SELECT user_id,is_admin,is_jurisdictional_admin,is_partner
+      FROM user_roles`
+    )
+
+    // Update user information
+    existingUsersAndRoles.forEach(async (user) => {
+      // Add all jurisdictions to the admin users
+      if (user.is_admin) {
+        existingJurisdictions.forEach(async (jurisdiction) => {
+          if (jurisdiction.name !== "Bay Area") {
+            await queryRunner.query(
+              `INSERT INTO user_accounts_jurisdictions_jurisdictions
+              (user_accounts_id, jurisdictions_id)
+              VALUES ($1, $2)`,
+              [user.user_id, jurisdiction.id]
+            )
+          }
+        })
+      // Add all jurisdictions tied to listings of partner accounts
+      } else if (user.is_partner) {
+        const userListings = await queryRunner.query(
+          `SELECT listings_id from listings_leasing_agents_user_accounts where user_accounts_id = '${user.user_id}'`
+        )
+        userListings.forEach(async (listing) => {
+          const matchingListing = existingListings.find(
+            (list) => list.id === listing.listings_id
+          )
+          const matchingJuris = existingJurisdictions.find((juris) => juris.name === matchingListing.county).id
+          console.log("matchingListing", matchingListing)
```

Comment:
Assuming we want to delete these @ludtkemorgan

Reply:
good catch. If you are ok with it. I'll approve it for you. Most of the code was done by you and I reviewed that portion

Reply:
Sounds good!